### PR TITLE
Linter

### PR DESCRIPTION
**Pull request by apd-ypuranik**
## Fixes/Addresses/Summary/Motivation:

#20 

## Changes proposed in this PR:
- Set up a pylint pre-commit hook. This will ensure all files newly updated must pass pylint before being committed
- Set up a pylint GitHub action. This ensures that globally the repo must pass a threshold score (currently set to 9) before a PR is allowed to merge
- Adds a citation to the repo

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my
contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.md file
   at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has
   rights to intellectual property that includes these contributions, I represent that I have
   received permission to make contributions and grant the required license on behalf of that
   employer.

### CODE REVIEW

**Comment by apd-ypuranik** [comment]
@ruonanli61 , @radhakrishnatg , @strahlw , @DevKakkad32 , @Jaffe11 : Please review and wait for #35 to be merged before addressing this one

File: .github/wordlist.txt [hunk 1]
```diff
@@ -41,36 +44,44 @@ csv
...
 ipynb
 ipywidgets
 iterable
 json
 libs
+msg
 multicommodity
 ndarray
 np
 num
```
Comment (radhakrishnatg) [inline .github/wordlist.txt:67]: Could you please add `numpy` to this list?

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    given-names: "Yash"
+  - family-names: "Kakkad"
+    given-names: "Dev"
+  - family-names: "Jaffe"
+    given-names: "Tyler"
+  - family-names: "Li"
+    given-names: "Ruonan"
+  - family-names: "Strahl"
+    given-names: "William"
+  - family-names: "Gooty"
```
Comment (radhakrishnatg) [inline CITATION.cff:14]: Could you please change the `family-names` to `"Tumbalam Gooty"` and `given-names` to `"Radhakrishna"`?

Comment (apd-ypuranik) [reply CITATION.cff:14]: happy to :-)

File: .github/wordlist.txt (hunk 1 above)
Comment (apd-ypuranik) [reply .github/wordlist.txt:67]: will do!

File: primo/data_parser/metric_data.py [hunk 3]
```diff
@@ -693,8 +691,14 @@ class ImpactMetrics(SetOfMetrics):
...
     def __init__(
         self,
-        impact_metrics: Optional[Dict[str, _SupportedContent]] = SUPP_IMPACT_METRICS,
+        impact_metrics: Optional[Dict[str, _SupportedContent]] = None,
     ) -> None:
+        # Default arguments retain values between function calls.
+        # If we use a mutable argument, unexpected things can occurs.
+        # See this for example:
+        # https://stackoverflow.com/questions/26320899/
+        if impact_metrics is None:
```
Comment (strahlw) [inline primo/data_parser/metric_data.py:700]: This post makes sense, but highlights that it is only a problem if the default argument is modified by the function. This is an input that is never modified, so I think we are safe. However, I don't see the change doing any harm.

File: primo/opt_model/tests/test_model_options.py [hunk 4]
```diff
@@ -154,7 +152,7 @@ def test_opt_model_inputs(get_column_names):
     assert "Clusters" in wd_gas
 
     opt_mdl_inputs.build_optimization_model()
-    opt_campaign = opt_mdl_inputs.solve_model(solver="scip")
+    opt_campaign = opt_mdl_inputs.solve_model(solver="highs")
     opt_mdl = opt_mdl_inputs.optimization_model
```
Comment (strahlw) [inline primo/opt_model/tests/test_model_options.py:156]: This reminds me we need to change the solver option for the API

**Review by strahlw — APPROVED** [review]

**Review by radhakrishnatg — APPROVED** [review]

File: primo/opt_model/tests/test_model_options.py (hunk 4 above)
Comment (apd-ypuranik) [reply primo/opt_model/tests/test_model_options.py:156]: Thanks Will!

File: primo/data_parser/metric_data.py (hunk 3 above)
Comment (apd-ypuranik) [reply primo/data_parser/metric_data.py:700]: Yep, just trying to avoid issues in the future if we make some changes to that...

**Pull request merged**